hack/update-codegen.sh

```diff
@@ -19,23 +19,36 @@ set -o nounset
 set -o pipefail
 
 SCRIPT_PACKAGE=${SCRIPT_PACKAGE:-"github.com/operator-framework/operator-metering"}
-SCRIPT_ROOT="$(realpath $(dirname ${BASH_SOURCE[0]})/..)"
-CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo k8s.io/code-generator)}
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../../../k8s.io/code-generator)}
+
+verify="${VERIFY:-}"
 
 set -x
+# Because go mod sux, we have to fake the vendor for generator in order to be able to build it...
+mv ${CODEGEN_PKG}/generate-groups.sh ${CODEGEN_PKG}/generate-groups.sh.orig
+sed 's/ go install/#go install/g' ${CODEGEN_PKG}/generate-groups.sh.orig > ${CODEGEN_PKG}/generate-groups.sh
+function cleanup {
+  mv ${CODEGEN_PKG}/generate-groups.sh.orig ${CODEGEN_PKG}/generate-groups.sh
+}
+trap cleanup EXIT
+
+go install ./${CODEGEN_PKG}/cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
```

Comment:
The more I think about this, the less I like this approach as we're essentially running `go install ...` on this package for every CI run which is slow (and as we've seen, prone to verify errors). 

I haven't tested this yet, but I wonder what's stopping us from copying the generate-groups.sh script from `vendor/k8s.io/code-generator/generate-groups.sh` and simply sourcing our hack/ version of that script in update-codegen.sh as a workaround to the "no such file" error I see in `make verify` errors.

Reply:
Well above in the current `update-codegen.sh` we have:

```bash
mv ${CODEGEN_PKG}/generate-groups.sh ${CODEGEN_PKG}/generate-groups.sh.orig
sed 's/ go install/#go install/g' ${CODEGEN_PKG}/generate-groups.sh.orig > ${CODEGEN_PKG}/generate-groups.sh
```
That's taking the `vendor/k8s.io/code-generator/generate-groups.sh` disabling `go install`, running `go install` ourselves (installing the generator commands) and then calling generate-groups as needed. When you say we're `go install...` this package, the this is only the generator commands, so not too bad, right?